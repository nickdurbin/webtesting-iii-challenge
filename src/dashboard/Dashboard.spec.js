import React from "react";
import * as rtl from "react-testing-library";
import Dashboard from './Dashboard';
import "jest-dom/extend-expect";

afterEach(rtl.cleanup);

test("<Dashboard /> snapshot", async () => {
  const wrapper = rtl.render(<Dashboard />)
  await wrapper.getByText(/locked/i);
  expect(wrapper.asFragment()).toMatchSnapshot()
})

test("Check open and closed functionality", async () => {
  const wrapper = rtl.render(<Dashboard />)
  await wrapper.getByText(/open/i);

  const closed = wrapper.getByText(/close gate/i);
  
  rtl.act(() => {
    rtl.fireEvent.click(closed)
  })

  expect(wrapper.getByText(/open/i)).toHaveTextContent(/open gate/i)
})

test("Check locked and unlocked functionality", async () => {
  const wrapper = rtl.render(<Dashboard />)
  await wrapper.getByText(/unlocked/i);

  const locked = wrapper.getByText(/locked/i);
  
  rtl.act(() => {
    rtl.fireEvent.click(locked)
  })

  expect(wrapper.getByText(/unlocked/i)).toHaveTextContent(/locked/i)
})