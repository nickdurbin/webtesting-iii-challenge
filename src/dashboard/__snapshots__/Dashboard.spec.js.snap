// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="lock"
      disabled=""
      role="button"
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="open"
      role="button"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
